src/environment.cc

```diff
@@ -164,6 +167,20 @@ const std::string& Environment::InstanceId() const {
   return instance_id_;
 }
 
+const std::string& Environment::InstanceResourceType() const {
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (instance_resource_type_.empty()) {
+    if (!config_.InstanceResourceType().empty()) {
+      instance_resource_type_ = config_.InstanceResourceType();
+    } else {
+      // Default to a GCE instance.
+      // TODO: Detect other instance resources.
+      instance_resource_type_ = kGceInstanceResourceType;
```

Comment:
Is there a reason that config_.InstanceResourceType cannot default to "gce_instance"?

Reply:
The current default (empty value) means "detect via the environment". I've added a comment in `configuration.cc` to that effect.